### ISSUE TITLE
Virtual fields in search index (feature request/question)

### ISSUE THREAD

**Issue by traed** [arangodb/arangodb#15829]
It would be great if there was a way to add fields to a certain index that does not exists on the original document when creating a view for arango search. For example, if I have a collection `humans` with documents like this:

```
{
  name: string,
  age: number
}
```
and an edge collection with:
```
{
  relationship_status: string
}
```

... I would like to be able to index and `SEARCH` for all humans with `name == 'John' AND relationship_status == 'married'`. So the indexed document would need to add the field from the edge collection at index time. I don't want to `FILTER` on `relationship_status` since that would be too slow.

Is there a way to achieve this currently or is it something that is possible to implement in the future?

**Comment by MBkkt** [comment]
Will be implemented in future, but probably not in 3.11

**Comment by MBkkt** [comment]
> Is there a way to achieve this currently 

Most similar is create field in collection and make it storedValues if necessary :(
